src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copy = {};
+
+  Object.assign(copy, state);
+
+  for (const i of actions) {
+    switch (i.type) {
+      case 'addProperties':
+        Object.assign(copy, i.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const j of i.keysToRemove) {
+          delete copy[j];
+        }
+        break;
+
+      case 'clear':
+        for (const k in copy) {
+          delete copy[k];
+        }
+        break;
+    }
```

Comment:
A `default` case should be added to the `switch` statement for error handling to cover any unexpected `type` values.